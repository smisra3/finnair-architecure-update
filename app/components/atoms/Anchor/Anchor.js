// @flow
import Link from 'next/link';
import type { Node } from 'react';
import { PureComponent } from 'react';
import styled from 'styled-components';

import styles from './Anchor.style';
import { buildUrl } from '../../../utils/utils';

type Props = {
  to: string | any,
  as?: string,
  children: Node,
  noLink?: boolean,
  className: '',
  inheritedStyles?: string,
  handleLinkClick: (SyntheticEvent<>) => void,
  shallow?: boolean,
  scroll?: boolean,
  forceClick?: boolean,
};

class Anchor extends PureComponent<Props> {
  /* eslint-disable */
  render() {
    const {
      children,
      to,
      as,
      className,
      noLink,
      inheritedStyles,
      handleLinkClick,
      shallow,
      scroll,
      forceClick,
      ...other
    } = this.props;

    return noLink ? (
      <a href={buildUrl(to)} className={className} onClick={handleLinkClick} {...other}>
        {children}
      </a>
    ) : (
      <Link href={to} as={as} shallow={shallow} scroll={scroll}>
        {
          forceClick ?
            <div>
              <a className={className} onClick={handleLinkClick} {...other}>
                {children}
              </a>
            </div>
            :
            <a className={className} {...other}>
              {children}
            </a>
        }
      </Link>
    );
  }
}

Anchor.defaultProps = {
  noLink: false,
  to: '',
  as: '',
  inheritedStyles: '',
  shallow: false,
  scroll: false,
  forceClick: false,
  handleLinkClick: 'function'
};

export default styled(Anchor)`${styles}`;

export { Anchor as AnchorVanilla };
