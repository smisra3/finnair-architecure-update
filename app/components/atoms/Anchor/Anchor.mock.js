import { action } from '@storybook/addon-actions';

const defaultAnchor = {
  className: 'example',
  to: 'https://nextjs.org/',
  handleLinkClick: action('clicked'),
};

const secondaryAnchor = {
  className: 'example',
  to: 'https://nextjs.org/',
  handleLinkClick: action('clicked'),
  secondary: true,
};

export {
  secondaryAnchor,
  defaultAnchor,
};
