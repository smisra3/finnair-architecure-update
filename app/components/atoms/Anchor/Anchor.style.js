import { css } from 'styled-components';

export default css`
  cursor: pointer;
  text-decoration: none;
  color: ${props => props.theme.linkColorPrimary};
  &:focus {
    outline: 1px dashed ${props => props.theme.colorOutline};
  }

  ${props =>
    (props.secondary
      ? `
        font-weight: ${props.theme.fontSemiBold}
        color: ${props.theme.linkColorSecondary}
      ` : '')};

  &:focus:not(.focus-visible) {
    outline: none;
  }
`;
