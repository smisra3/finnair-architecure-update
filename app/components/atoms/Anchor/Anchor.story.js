// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {defaultAnchor, secondaryAnchor} from './Anchor.mock';
// Import Vanilla Component to display markup and props
import { AnchorVanilla } from './Anchor';
// Import Styled Component to showcase variations
import Anchor from './index';

storiesOf('Atoms', module).addWithChapters('Anchor', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <AnchorVanilla {...defaultAnchor} className={`hide-default-sample ${defaultAnchor.className}`}>Home</AnchorVanilla>,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary Anchor Variations',
      sections: [
        {
          title: 'Default Link',
          sectionFn: () => <Anchor {...defaultAnchor}>Submit</Anchor>,
        },
        {
          title: 'Secondary Link',
          sectionFn: () => <Anchor {...secondaryAnchor}>Submit</Anchor>,
        },
      ],
    },
  ],
});
