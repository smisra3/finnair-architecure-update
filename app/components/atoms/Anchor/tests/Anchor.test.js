import Router from 'next/router';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Anchor from '../';
import { AnchorVanilla } from '../Anchor';

describe('Anchor Component', () => {
  let anchor;

  test('should render correctly and match snapshot', () => {
    anchor = shallow(<Anchor to="/sample">Sample</Anchor>);
    expect(anchor).toMatchSnapshot();
  });

  test('should render with styles correctly and match snapshot', () => {
    anchor = renderer.create(<Anchor to="/sample">Sample</Anchor>).toJSON();
    expect(anchor).toMatchSnapshot();
  });

  test('Should handle on typeahead focuson Events', () => {
    const mockedRouter = { push: () => {} };
    Router.router = mockedRouter;
    const props = { noLink: true, to: '/sample', handleLinkClick: jest.fn() };
    const event = { preventDefault: jest.fn() };
    anchor = shallow(<AnchorVanilla {...props}>Sample</AnchorVanilla>);
    anchor.find('a').simulate('click', event);
    expect(props.handleLinkClick).toHaveBeenCalled();
  });
});
