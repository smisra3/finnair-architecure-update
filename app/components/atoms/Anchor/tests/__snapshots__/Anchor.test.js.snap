// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component should render correctly and match snapshot 1`] = `
.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed;
}

.c0:focus:not(.focus-visible) {
  outline: none;
}

<Anchor
  as=""
  className="c0"
  forceClick={false}
  handleLinkClick="function"
  inheritedStyles=""
  noLink={false}
  scroll={false}
  shallow={false}
  to="/sample"
>
  Sample
</Anchor>
`;

exports[`Anchor Component should render with styles correctly and match snapshot 1`] = `
.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed;
}

.c0:focus:not(.focus-visible) {
  outline: none;
}

<a
  className="c0"
  href="/sample"
  onClick={[Function]}
>
  Sample
</a>
`;
